### PR TITLE
feat: add support for Jotai V2 API

### PR DESCRIPTION
**Pull request by arjunvegda**
- Add V2 API support
- Deprecate `modules` in `package.json`

TODO
- [x] Update docs

Tested it using the latest version of `jotai` and works as expected

### CODE REVIEW

**Review by dai-shi — DISMISSED** [review]
Looking forward to it!

File: README.md [hunk 1]
```diff
@@ -34,11 +38,19 @@ import {
 } from 'jotai-devtools';
 ```
 
-### Migration guide from `jotai/devtools` to `jotai-devtools`
+### Migration guide from `jotai/react/devtools` to `jotai-devtools`
```
Comment (dai-shi) [inline README.md:41]: Note that, once this is (pre)released, I will drop `jotai/react/devtools`.

**Review by dai-shi — APPROVED** [review]
(I didn't check the build result, but the commit message seems good.)

File: README.md (hunk 1 above)
Comment (arjunvegda) [reply README.md:46]: That makes sense! I'll update the docs once Jotai `v2` is released then.

**Pull request merged**